scipy/_build_utils/src/scipy_dll.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#if defined(SCIPY_DLL_EXPORTS)
+    #define SCIPY_DLL __declspec(dllexport)
+#elif defined(SCIPY_DLL_IMPORTS)
+    #define SCIPY_DLL __declspec(dllimport)
```

Comment:
https://github.com/conda-forge/scipy-feedstock/pull/277#issuecomment-2140839770 contains a detailed example with code comments that would be useful to take over here.